src/hooks/useStripeSetupIntent.tsx

```diff
@@ -0,0 +1,37 @@
+import { useCallback, useContext } from 'react';
+import { TokenContext } from 'hooks/useCustomerToken';
+import { useAsyncOnDemand } from 'hooks/useAsync';
+import type { UseAsyncOnDemandReturnType } from 'hooks/useAsync';
+import { createStripeSetupIntent } from 'api/octane';
+import { components } from '../apiTypes';
+
+export type StripeSetupIntent =
+  components['schemas']['CustomerPortalStripeCredential'];
+
+const useStripeSetupIntent = (args?: {
+  token?: string;
+}): UseAsyncOnDemandReturnType<StripeSetupIntent> => {
+  const { token: tokenFromContext } = useContext(TokenContext);
+  const userToken = args?.token || tokenFromContext;
+
+  if (!userToken) {
+    throw new Error('Token must be provided.');
+  }
+
+  const mutation = useCallback(() => {
+    return createStripeSetupIntent({
```

Comment:
Rather than using a `then` promise chain, use async/await. It's cleaner, and it makes it easier to do multi-step things without worrying about `then` chaining correctly.

Reply:
Rewrited on async/await style